nix_update/eval.py

```diff
@@ -117,7 +117,7 @@ def eval_attr(opts: Options) -> Package:
         "nix-command",
         "--expr",
         expr,
-    ]
+    ] + opts.system_flags
```

Comment:
I don't think this would actually work the way you want. By passing `--system` to nix, it makes the current nix instance believe it is a $system now and attempt to build packages for other operating systems/cpu architectures. The correct way however is setting `localSystem` when importing nixpkgs (not needed for flakes as this is already correctly set when exposing packages). This will make sure that a remote builder is used whenever a package needs to be build that does not match the current system.

Reply:
Can you elaborate on the differences? pretty sure `localSystem` [defaults](https://github.com/NixOS/nixpkgs/blob/9813adc7f7c0edd738c6bdd8431439688bb0cb3d/pkgs/top-level/impure.nix#L17) to `builtins.currentSystem`, which is [equilvalent](https://github.com/NixOS/nix/blob/d02c5a41da68908bd88d7a697abc52a04826894e/src/libexpr/primops.cc#L3984-L3985) to `--system` [option](https://github.com/NixOS/nix/blob/d02c5a41da68908bd88d7a697abc52a04826894e/src/libstore/globals.hh#L157)

Reply:
localSystem defaults to builtins.currentSystem. However changing it to a different value is not the same as changing the system of the locally running nix. When you change the system of nix it tries to build aarch64-linux binaries on an x86-64-linux:

```
$ nix-build --system 'aarch64-linux' --builders '' -E "with import <nixpkgs> {}; pkgs.writeText \"foo\" ''Test''"
ErrorErrorvexecuting '/nix/store/hjala5j03vxhkj6rvgj8kdigm1yrw0p4-bash-5.1-p16/bin/bash': Exec format error
error: builder for '/nix/store/m16gsw4qlb6l2hqqnsb7dy74kldbv54i-foo.drv' failed with exit code 1;
       last 2 log lines:
       > 
       > ErrorErrorvexecuting '/nix/store/hjala5j03vxhkj6rvgj8kdigm1yrw0p4-bash-5.1-p16/bin/bash': Exec format error
       For full logs, run 'nix log /nix/store/m16gsw4qlb6l2hqqnsb7dy74kldbv54i-foo.drv'.
```

Whereas if you change system than it will use a remote builder if available:

```
$  nix-build --builders '' -E "with import <nixpkgs> { system = \"aarch64-linux\"; }; pkgs.writeText \"foo\" ''Test''"
this derivation will be built:
  /nix/store/m16gsw4qlb6l2hqqnsb7dy74kldbv54i-foo.drv
error: a 'aarch64-linux' with features {} is required to build '/nix/store/m16gsw4qlb6l2hqqnsb7dy74kldbv54i-foo.drv', but I am a 'x86_64-linux' with features {benchmark, big-parallel, gccarch-broadwell, gccarch-haswell, gccarch-ivybridge, gccarch-san
dybridge, gccarch-skylake, gccarch-westmere, gccarch-znver1, gccarch-znver2, gccarch-znver3, kvm, nixos-test}
```

Reply:
I can't reproduce that, but anyways, I have added that to the import inputs
```
$ nix-build --system 'aarch64-linux' --builders '' -E "with import <nixpkgs> {}; pkgs.writeText \"foo\" ''Test''"
this derivation will be built:
  /nix/store/q0x61xkb7139ksc9dn3sc9n26sr0nbai-foo.drv
error: a 'aarch64-linux' with features {} is required to build '/nix/store/q0x61xkb7139ksc9dn3sc9n26sr0nbai-foo.drv', but I am a 'x86_64-linux' with features {benchmark, big-parallel, kvm, nixos-test}
```